torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -194,14 +194,45 @@ Buffer texprBuffer(const torch::jit::Value* v) {
 }
 
 template <typename T>
-size_t bufferSize(T t) {
-  size_t size = 1;
+int64_t bufferSize(T t) {
+  int64_t size = 1;
   for (int i = 0; i < t.ndim(); i++) {
     size *= t.dim(i).template AsNode<IntImm>()->value();
   }
   return size;
 }
 
+template <typename T>
+std::vector<int64_t> bufferSizes(const T& t) {
+  std::vector<int64_t> sizes;
+  for (int i = 0; i < t.ndim(); i++) {
+    sizes.push_back(t.dim(i).template AsNode<IntImm>()->value());
+  }
+  return sizes;
+}
+
+template <typename T>
+std::vector<Expr> broadcastArgs(
```

Comment:
Minor nitpicking: Maybe a better name to reflect this flows from output-index and input-shapes to input-index? Like "InputIndexFromOutputDims". For example: "axes"->"output_indices", "sizes" -> "input_sizes", return "input_indices"?

Reply:
Yeah maybe a good idea.  I kinda want "broadcast" in the name since the point of the function is to implement broadcast semantics.  But the way it does it is a little odd.  Maybe renaming the args as you suggested is sufficient?  Or maybe something like `computeAxesToBroadcast`?